src/scikit_build_core/builder/builder.py

```diff
@@ -227,7 +227,11 @@ def configure(
             cmake_args=[*self.get_cmake_args(), *configure_args],
         )
 
-    def build(self, build_args: list[str]) -> None:
+    def build(self, build_args: Sequence[str]) -> None:
+        build_tool_args = self.settings.build.tool_args
+        if build_tool_args:
+            build_args = [*build_args, "--", *build_tool_args]
```

Comment:
@henryiii Is `build_args` here the CMake args list? I'm [seeing](https://github.com/rapidsai/cudf/actions/runs/9086894250/job/24973580680?pr=15753#step:7:5505) strange behavior where the ninja args are wrapped in a string:
```
*** Building project with Default Generator...
2024-05-14 22:28:14,653 - scikit_build_core - INFO - RUN: .local/share/venvs/rapids/lib/python3.10/site-packages/cmake/data/bin/cmake --build cudf/python/cudf/build/pip/cuda-12.2/release -v -- -v -j8 -l8
Change Dir: 'cudf/python/cudf/build/pip/cuda-12.2/release'
Run Build Command(s): .local/share/venvs/rapids/lib/python3.10/site-packages/ninja/data/bin/ninja -v -j 8 "-v -j8 -l8"
.local/share/venvs/rapids/lib/python3.10/site-packages/ninja/data/bin/ninja: invalid option -- ' '
```

It's possible there's multiple things going on. I have both `CMAKE_BUILD_PARALLEL_LEVEL=8` and `SKBUILD_BUILD_TOOL_ARGS="-v -j8 -l8"`, so it looks like `-j8` is duplicated.

I don't believe `-v -j8 -l8` should be a single shell arg, but I'm not sure why `cmake --build <dir> -- -v -j8 -l8` would be passing it as one unless everything after the `--` is a single shell argument?

Reply:
I think it's `SKBUILD_BUILD_TOOL_ARGS` parsing. `config_settings` seem to be working just fine:

```
$ python -m gpep517 build-wheel --output-fd 1 --wheel-dir dist --config-json '{"build.tool-args": ["-j12", "-l12"]}'
```

Reply:
You only need the `-l8`, the others can be set from CMake or scikit-build-core.

Reply:
At a computer now. All `SKBUILD_*` vars are `;` separated, like cmake. So it's `SKBUILD_BUILD_TOOL_ARGS="-v;-j8;-l8"`. But you don't need the first two; you can do `CMAKE_BUILD_PARALLEL_LEVEL=8 SKBUILD_CMAKE_VERBOSE=1 SKBUILD_BUILD_TOOL_ARGS=-l8`. Or, pass via config settings with `cmake.verbose` and `build.tool-args`. For pip and build, you make a list by passing multiple times, so `pip install . -Cbuild.tool-args="-j12" -Cbuild.tool-args="-l12" -Cbuild.tool-args="-v"` would be how you'd do it.

I'll probably rename cmake.verbose to build.verbose now that we have a build section. The old one will work though for back-compat.